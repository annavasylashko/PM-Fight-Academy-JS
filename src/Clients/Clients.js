import React from "react";
import styles from "./clients.module.css";

class Clients extends React.Component {
  render() {
    return <h1>Clients</h1>;
  }
}

export default Clients;
