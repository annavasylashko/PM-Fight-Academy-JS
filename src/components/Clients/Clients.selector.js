export const clientsSelector = (state) => state.client.data;

// export default clientsSelector;
