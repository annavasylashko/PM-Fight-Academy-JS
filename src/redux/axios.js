import axios from "axios";

export default axios.create({
  baseURL: "https://pmfightacademyadmin-hr8mp.ondigitalocean.app/",
});
