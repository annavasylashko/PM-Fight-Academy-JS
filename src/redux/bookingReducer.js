import {
  GET_BOOKINGS,
  RESET_BOOKINGS,
  DELETE_BOOKING,
  UPDATE_BOOKING,
} from "./types";

const initialState = {
  data: [],
};

const handlers = {
  [GET_BOOKINGS]: (state, { payload }) => ({
    ...state,
    data: [...state.data, ...payload],
  }),
  [RESET_BOOKINGS]: (state) => initialState,
  DEFAULT: (state) => state,
  [UPDATE_BOOKING]: (state, { payload }) => ({
    ...state,
    data: [
      ...state.data.map((item) => (item.id === payload.id ? payload : item)),
    ],
  }),
  [DELETE_BOOKING]: (state, { payload }) => ({
    ...state,
    data: [...state.data.filter((item) => item.id !== payload)],
  }),
};

export const bookingReducer = (state = initialState, action) => {
  const handle = handlers[action.type] || handlers.DEFAULT;
  return handle(state, action);
};
