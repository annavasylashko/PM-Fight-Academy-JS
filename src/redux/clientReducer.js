import { GET_CLIENTS, RESET_CLIENTS } from "./types";

const initialState = {
  data: [],
};

const handlers = {
  [GET_CLIENTS]: (state, { payload }) => ({
    ...state,
    data: [...state.data, ...payload],
  }),
  [RESET_CLIENTS]: (state) => initialState,
  DEFAULT: (state) => state,
};

export const clientReducer = (state = initialState, action) => {
  const handle = handlers[action.type] || handlers.DEFAULT;
  return handle(state, action);
};
