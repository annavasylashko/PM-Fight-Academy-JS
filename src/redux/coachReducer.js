import {
  ADD_COACH,
  GET_COACHES,
  UPDATE_COACH,
  DELETE_COACH,
  RESET_COACHES,
} from "./types";

const initialState = {
  data: [],
};

const handlers = {
  [GET_COACHES]: (state, { payload }) => ({
    ...state,
    data: [...state.data, ...payload],
  }),
  [ADD_COACH]: (state, { payload }) => ({
    ...state,
    data: [...state.data, payload],
  }),
  [UPDATE_COACH]: (state, { payload }) => ({
    ...state,
    data: [
      ...state.data.map((item) => (item.id === payload.id ? payload : item)),
    ],
  }),
  [DELETE_COACH]: (state, { payload }) => ({
    ...state,
    data: [...state.data.filter((item) => item.id !== payload)],
  }),
  [RESET_COACHES]: (state) => initialState,
  DEFAULT: (state) => state,
};

export const coachReducer = (state = initialState, action) => {
  const handle = handlers[action.type] || handlers.DEFAULT;
  return handle(state, action);
};
