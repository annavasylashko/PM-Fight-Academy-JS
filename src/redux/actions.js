import {
  ADD_SERVICE,
  GET_SERVICES,
  RESET_SERVICES,
  ADD_COACH,
  GET_COACHES,
  UPDATE_COACH,
  DELETE_COACH,
  RESET_COACHES,
  GET_CLIENTS,
  RESET_CLIENTS,
  UPDATE_SERVICE,
  DELETE_SERVICE,
  GET_SLOTS,
  ADD_SLOT,
  UPDATE_SLOT,
  DELETE_SLOT,
  RESET_SLOTS,
  GET_QUALIFICATIONS,
  ADD_QUALIFICATION,
  DELETE_QUALIFICATION,
  RESET_QUALIFICATIONS,
  GET_BOOKINGS,
  RESET_BOOKINGS,
  DELETE_BOOKING,
} from "./types";
import {mockApi as Api, realApi as Api2} from "./axios";

//

export function getServices() {
  return async (dispatch) => {
    const services = await Api.get("/Services");
    dispatch({
      type: GET_SERVICES,
      payload: services.data,
    });
  };
}

export function addService(service) {
  return async (dispatch) => {
    try {
      const response = await Api.post("/Services", service);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: ADD_SERVICE,
      payload: service,
    });
  };
}

export function updateService(service) {
  return async (dispatch) => {
    try {
      const response = await Api.patch(`Services/${service.id}`, service);

    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: UPDATE_SERVICE,
      payload: service,
    });
  };
}

export function deleteService(serviceId) {
  return async (dispatch) => {
    try {
      const response = await Api.delete(`Services/${serviceId}`);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: DELETE_SERVICE,
      payload: serviceId,
    });
  };
}

export function resetService() {
  return (dispatch) => {
    dispatch({
      type: RESET_SERVICES,
    });
  };
}

//

export function getCoaches() {
  return async (dispatch) => {
    const coaches = await Api.get("Coach/");
    dispatch({
      type: GET_COACHES,
      payload: coaches.data,
    });
  };
}

export function addCoach(coach) {
  return async (dispatch) => {
    try {
      const response = await Api.post("Coach/", coach);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: ADD_COACH,
      payload: coach,
    });
  };
}

export function updateCoach(coach) {
  return async (dispatch) => {
    try {
      const response = await Api.patch(`Coach/update/${coach.id}`, coach);
    } catch (e) {
      console.log(e.message);
    }
    dispatch({
      type: UPDATE_COACH,
      payload: coach,
    });
  };
}

export function deleteCoach(coachId) {
  return async (dispatch) => {
    try {
      const response = await Api.delete(`Service/${coachId}`);
    } catch (e) {
      console.log(e.message);
    }
    dispatch({
      type: DELETE_COACH,
      payload: coachId,
    });
  };
}

export function resetCoach() {
  return (dispatch) => {
    dispatch({
      type: RESET_COACHES,
    });
  };
}

//

export function getClients() {
  return async (dispatch) => {
    const clients = await Api.get("Client/");
    dispatch({
      type: GET_CLIENTS,
      payload: clients.data,
    });
  };
}

export function resetClient() {
  return (dispatch) => {
    dispatch({
      type: RESET_CLIENTS,
    });
  };
}

//

export function getSlots() {
  return async (dispatch) => {
    const slots = await Api.get("Slots/");
    dispatch({
      type: GET_SLOTS,
      payload: slots.data,
    });
  };
}

export function addSlot(slot) {
  return async (dispatch) => {
    try {
      const response = await Api.post("Slots/", slot);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: ADD_SLOT,
      payload: slot,
    });
  };
}

export function updateSlot(slot) {
  return async (dispatch) => {
    try {
      const response = await Api.patch(`Slots/update/${slot.id}`, slot);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: UPDATE_SLOT,
      payload: slot,
    });
  };
}

export function deleteSlot(slotId) {
  return async (dispatch) => {
    try {
      const response = await Api.delete(`Slots/${slotId}`);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: DELETE_SLOT,
      payload: slotId,
    });
  };
}

export function resetSlots() {
  return (dispatch) => {
    dispatch({
      type: RESET_SLOTS,
    });
  };
}

//

//BOOKING

//

export function getQualifications(param, id) {
  return async (dispatch) => {
    const qualifications = await Api.get(`Qualification?${param}=${id}`);
    dispatch({
      type: GET_QUALIFICATIONS,
      payload: qualifications.data,
    });
  };
}

export function addQualification(qualification) {
  return async (dispatch) => {
    try {
      const response = await Api.post("Qualification/", qualification);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: ADD_QUALIFICATION,
      payload: qualification,
    });
  };
}

export function deleteQualifications(qualificationId) {
  return async (dispatch) => {
    try {
      const response = await Api.delete(`Qualification/${qualificationId}`);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: DELETE_QUALIFICATION,
      payload: qualificationId,
    });
  };
}

export function resetQualifications() {
  return (dispatch) => {
    dispatch({
      type: RESET_QUALIFICATIONS,
    });
  };
}

export function getBookings() {
  return async (dispatch) => {
    const booking = await Api.get("Booking/");
    dispatch({
      type: GET_BOOKINGS,
      payload: booking.data,
    });
  };
}

export function resetBookings() {
  return (dispatch) => {
    dispatch({
      type: RESET_BOOKINGS,
    });
  };
}

export function deleteBooking(bookingId) {
  return async (dispatch) => {
    try {
      const response = await Api.delete(`Booking/${bookingId}`);
    } catch (e) {
      console.log(e.message);
    }

    dispatch({
      type: DELETE_BOOKING,
      payload: bookingId,
    });
  };
}
