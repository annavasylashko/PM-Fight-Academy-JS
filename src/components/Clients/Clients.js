import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getClients, resetClient } from "../../redux/actions";

import styles from "../MainStyles/mainStyles.module.css";

const Clients = () => {
  const dispatch = useDispatch();

  const clients = useSelector((state) => state.client.data);

  useEffect(() => {
    dispatch(getClients());
    return () => dispatch(resetClient());
  }, []);

  const formatClients = () =>
    clients?.map((client) => (
      <div key={client.id}>
        <p>{client.name}</p>
        <span className={styles.clientPhoneNumber}>{client.phoneNumber}</span>
      </div>
    ));

  return (
    <div className={styles.clientContainer}>
      <div className={styles.resultList}>{formatClients()}</div>
    </div>
  );
};

export default Clients;
