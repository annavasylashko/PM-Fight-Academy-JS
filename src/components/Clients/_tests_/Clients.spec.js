import React from "react";
import { mount, shallow } from "enzyme";
import Clients from "../Clients";
import clientsSelector from "../Clients.selector";

import { getClients, resetClient } from "../../../redux/actions";
import { useDispatch, useSelector } from "react-redux";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useDispatch: () => mockDispatch,
  useSelector: (selector) => selector,
}));

jest.mock("../../../redux/actions", () => ({
  getClients: jest.fn(() => Symbol.for("getClients")),
  resetClient: jest.fn(() => Symbol.for("resetClient")),
}));

jest.mock("../Clients.selector", () => ({
  __esModule: true,
  default: jest.fn(() => []),
}));

describe("Clients component", () => {
  beforeEach(() => {
    getClients.mockReturnValue(Symbol.for("getClients"));
    resetClient.mockReturnValue(Symbol.for("resetClients"));
  });

  it("should get clients list on mount and reset list unmount", () => {
    const component = mount(<Clients />);
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch).toHaveBeenCalledWith(Symbol.for("getClients"));
    component.unmount();
    // expect(jest.fn()).toHaveBeenCalledTimes(2);
    // expect(mockDispatch).toHaveBeenCalledWith(Symbol.for("resetClients"));
  });
  it("should render list of clients", () => {
    clientsSelector.mockReturnValueOnce(() => [
      {
        id: 1,
        name: "Joni",
        phoneNumber: "911",
      },
      {
        id: 2,
        name: "Julia",
        phoneNumber: "101",
      },
      {
        id: 3,
        name: "Julian",
        phoneNumber: "1287678601",
      },
    ]);

    const component = shallow(<Clients />);

    // const resultList = component.find(".resultList");
    // expect(resultList.children.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
  // it("should render empty page if there are no clients in the list", () => {
  //   clientsSelector.mockReturnValueOnce([]);
  //   const component = shallow(<Clients />);

  //   const resultList = component.find(".resultList");
  //   expect(resultList.children.length).toBe(0);
  //   expect(component).toMatchSnapshot();
  // });
});
