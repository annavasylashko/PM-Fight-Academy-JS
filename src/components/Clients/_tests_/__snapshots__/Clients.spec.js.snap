// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clients component should render list of clients 1`] = `
<div
  className="clientContainer"
>
  <div
    className="resultList"
  />
</div>
`;
